compiler/noirc_frontend/src/tests.rs

```diff
@@ -1572,6 +1593,22 @@ fn numeric_generic_as_struct_field_type_fails() {
     check_errors(src);
 }
 
+#[test]
+fn type_alias_to_numeric_generic() {
+    let src = r#"
+    type Double<let N: u32>: u32 = N * 2;
+    fn main() {
+        let b: [u8; 6] = foo();
+        assert(b[0] == 0);
+    }
+    fn foo<let N:u32>() -> [u8;Double::<N>] {
```

Comment:
Can we parse `Double::<N>` instead as `Double<N>`? That way it'd fit more in line with generics on other non-numeric type aliases

Reply:
Hmm, I see the problem though. We're parsing an expression here not a type so we run into something like `D < N > 2` actually being a valid expression...